### PR TITLE
Restructure Makefiles

### PR DESCRIPTION
**Pull request by AndreasFuchsSIT**
This is an attempt to make our Makefiles more readable and easier maintainable.

### CODE REVIEW

File: Makefile-test.am [hunk 1]
```diff
@@ -0,0 +1,471 @@
...
+# IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE 
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE 
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR 
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF 
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS 
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN 
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) 
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF 
+# THE POSSIBILITY OF SUCH DAMAGE.
+#;**********************************************************************;
```
Comment (flihp) [inline Makefile-test.am]: There are a lot of whitespace errors in the license header. Likely they were there in the original Makefile.am and they just got copied over. We should keep this from spreading.

File: Makefile.am [hunk 2]
```diff
@@ -24,255 +25,122 @@
...
 #;**********************************************************************;
-@CODE_COVERAGE_RULES@
-@DX_RULES@
 
-MOSTLYCLEANFILES = $(DX_CLEANFILES)
-
-include src_vars.mk
 
+### Initialize global variables used throughout the file ###
+# Initialize variables used throughout
```
Comment (flihp) [inline Makefile.am]: These comments are redundant. Think we can get away with just one comment?

File: Makefile.am [hunk 3]
```diff
@@ -24,255 +25,122 @@
...
 
-# stuff to build, what that stuff is, and where/if to install said stuff
-lib_LTLIBRARIES = $(libmarshal) $(libsapi) $(libtcti_device) $(libtcti_socket)
-noinst_LTLIBRARIES = $(libutil)
+# Initialize empty variables to be extended throughout
+lib_LTLIBRARIES =
+noinst_LTLIBRARIES =
+EXTRA_DIST =
+CLEANFILES =
+MOSTLYCLEANFILES =
```
Comment (flihp) [inline Makefile.am]: whitespace error at the end of this line

File: Makefile.am [hunk 4]
```diff
@@ -24,255 +25,122 @@
...
+### Add ax_* rules ###
+# ax_code_coverage
+@CODE_COVERAGE_RULES@
+
+# ax_doxygen
+@DX_RULES@
+MOSTLYCLEANFILES += $(DX_CLEANFILES)
+
+
+
```
Comment (flihp) [inline Makefile.am]: Lots of unnecessary new lines here.

File: Makefile.am [hunk 5]
```diff
@@ -24,255 +25,122 @@
...
-    lib/tcti-device.pc \
-    lib/tcti-socket.pc
-# man pages / documentation
-man3_MANS = man/man3/InitDeviceTcti.3 man/man3/InitSocketTcti.3
-man7_MANS = man/man7/tcti-device.7 man/man7/tcti-socket.7
 
-EXTRA_DIST = \
-    AUTHORS \
+### Distributiuon files ###
+# EXTRA_DIST Files
```
Comment (flihp) [inline Makefile.am]: I'm not sure these comments add any useful data. If you think the EXTRA_DIST variable needs documentation we should add what it does: include these files in the release tarball.

File: Makefile.am [hunk 6]
```diff
@@ -282,322 +150,82 @@ tcti_libtcti_socket_la_LIBADD   = $(libmarshal) $(URIPARSER_LIBS) $(libutil)
...
+sysapi_libsapi_la_SOURCES = $(SYSAPI_C) $(SYSAPI_H) $(SYSAPIUTIL_C) \
+    $(SYSAPIUTIL_H)
+
+nodist_pkgconfig_DATA += lib/sapi.pc
+
+EXTRA_DIST += \
+    lib/sapi.pc.in
+
+
+
```
Comment (flihp) [inline Makefile.am]: More unnecessary whitespace here ...

File: Makefile.am [hunk 7]
```diff
@@ -607,3 +235,4 @@ define make_man
     cat $2 $3 > $1 && \
     sed -i -e "s,[@]VERSION[@],$(PACKAGE_VERSION),g;" $1
 endef
+
```
Comment (flihp) [inline Makefile.am]: another whitespace error

File: Makefile.am [hunk 8]
```diff
@@ -282,322 +150,82 @@ tcti_libtcti_socket_la_LIBADD   = $(libmarshal) $(URIPARSER_LIBS) $(libutil)
...
+    man/InitSocketTcti.3.in \
+    man/tcti-device.7.in \
+    man/tcti-socket.7.in
+
+CLEANFILES += \
+    $(man3_MANS) \
+    $(man7_MANS)
 
+
+
```
Comment (flihp) [inline Makefile.am]: another unnecessary block of newlines, ones enough

File: Makefile.am [hunk 9]
```diff
@@ -282,322 +150,82 @@ tcti_libtcti_socket_la_LIBADD   = $(libmarshal) $(URIPARSER_LIBS) $(libutil)
...
+esapi_libesapi_la_SOURCES = $(ESAPI_SRC)
+endif #ESAPI
 
-%.pc : %.pc.in
-	$(AM_V_GEN)$(call make_parent_dir,$@) && \
-	sed -e "s,[@]VERSION[@],$(PACKAGE_VERSION),g; \
-	        s,[@]libdir[@],$(libdir),g; \
-	        s,[@]includedir[@],$(includedir),g;" $^ > $@
+
+
```
Comment (flihp) [inline Makefile.am]: another block of unnecessary newlines

**Comment by AndreasFuchsSIT** [comment]
@flihp Quick question: Do you know what the point of the file lib/debug_config.site ?

Also, I added the 3 newlines for readability. I found it way easier to fine the respective blocks.
So I'd vote for keeping them ?

File: Makefile-test.am [hunk 10]
```diff
@@ -0,0 +1,471 @@
...
+test_integration_create_loaded_int_LDADD   = $(TESTS_LDADD)
+test_integration_create_loaded_int_SOURCES = test/integration/main.c \
+    test/integration/create-loaded.int.c
+
+test_integration_policy_authorizeNV_int_CFLAGS  = $(AM_CFLAGS) $(TESTS_CFLAGS)
+test_integration_policy_authorizeNV_int_LDADD   = $(TESTS_LDADD)
+test_integration_policy_authorizeNV_int_SOURCES = test/integration/main.c log/log.h log/log.c \
+    test/integration/policy-authorizeNV.int.c
+endif #SIMULATOR_BIN
+
```
Comment (flihp) [inline Makefile-test.am]: sorry @AndreasFuchsSIT, another new line that we don't want

File: Makefile.am [hunk 11]
```diff
@@ -1,278 +1,135 @@
...
+libtcti_device_la_CFLAGS   = $(AM_CFLAGS)
 if HAVE_LD_VERSION_SCRIPT
-tcti_libtcti_device_la_LDFLAGS  = -Wl,--version-script=$(srcdir)/tcti/tcti_device.map
+libtcti_device_la_LDFLAGS  = -Wl,--version-script=$(srcdir)/tcti/tcti_device.map
 endif # HAVE_LD_VERSION_SCRIPT
-tcti_libtcti_device_la_LIBADD   = $(libmarshal) $(libutil)
-tcti_libtcti_device_la_SOURCES  = tcti/tcti_device.c
+libtcti_device_la_LIBADD   = $(libmarshal) $(libutil)
+libtcti_device_la_SOURCES  = tcti/tcti_device.c
```
Comment (flihp) [inline Makefile.am]: Changing this from `tcti_libtcti_device_la` to `libtcti_device_la` moves the location of the `libtcti-device.la` file and where the `libtcti-device.so` will end up. Was this intentional? If so some justification would be helpful.

File: Makefile-test.am (hunk 10 above)
Comment (AndreasFuchsSIT) [reply Makefile-test.am]: done

File: Makefile.am (hunk 11 above)
Comment (AndreasFuchsSIT) [reply Makefile.am]: Sorry, that was testing stuff... Reverted to original...
